Source/Core/DiscIO/VolumeGC.cpp

```diff
@@ -92,19 +96,131 @@ int CVolumeGC::GetRevision() const
 	return revision;
 }
 
-std::vector<std::string> CVolumeGC::GetNames() const
+std::string CVolumeGC::GetName() const
 {
-	std::vector<std::string> names;
-
-	auto const string_decoder = GetStringDecoder(GetCountry());
-
 	char name[0x60 + 1] = {};
 	if (m_pReader != nullptr && Read(0x20, 0x60, (u8*)name))
-		names.push_back(string_decoder(name));
+		return DecodeString(name);
+	else
+		return "";
+}
+
+std::vector<std::pair<IVolume::ELanguage, std::string>> CVolumeGC::GetNames() const
+{
+	std::vector<std::pair<IVolume::ELanguage, std::string>> names;
+
+	if (!LoadBannerFile())
+		return names;
+
+	u32 name_count = 0;
+	IVolume::ELanguage language;
+	bool is_japanese = GetCountry() == IVolume::ECountry::COUNTRY_JAPAN;
+
+	switch (m_banner_file_type)
+	{
+	case BANNER_BNR1:
+		name_count = 1;
+		language = is_japanese ? IVolume::ELanguage::LANGUAGE_JAPANESE : IVolume::ELanguage::LANGUAGE_ENGLISH;
+		break;
+
+	case BANNER_BNR2:
+		name_count = 6;
+		language = IVolume::ELanguage::LANGUAGE_ENGLISH;
+		break;
+
+	default:
+		break;
+	}
+
+	auto const banner = reinterpret_cast<const GCBanner*>(m_banner_file.data());
+
+	for (u32 i = 0; i < name_count; ++i)
+	{
+		auto& comment = banner->comment[i];
+		std::string name = DecodeString(comment.longTitle);
+
+		if (name.empty())
+			name = DecodeString(comment.shortTitle);
+
+		if (!name.empty())
+			names.push_back(std::pair<IVolume::ELanguage, std::string>((IVolume::ELanguage)(language + i), name));
+	}
 
 	return names;
 }
 
+std::vector<std::pair<IVolume::ELanguage, std::string>> CVolumeGC::GetDescriptions() const
+{
+	std::vector<std::pair<IVolume::ELanguage, std::string>> descriptions;
+
+	if (!LoadBannerFile())
+		return descriptions;
+
+	u32 desc_count = 0;
+	IVolume::ELanguage language;
+	bool is_japanese = GetCountry() == IVolume::ECountry::COUNTRY_JAPAN;
+
+	switch (m_banner_file_type)
+	{
+	case BANNER_BNR1:
+		desc_count = 1;
+		language = is_japanese ? IVolume::ELanguage::LANGUAGE_JAPANESE : IVolume::ELanguage::LANGUAGE_ENGLISH;
+		break;
+
+	case BANNER_BNR2:
+		language = IVolume::ELanguage::LANGUAGE_ENGLISH;
+		desc_count = 6;
+		break;
+
+	default:
```

Comment:
default should be replaced with
```cpp
case BANNER_NOT_LOADED:
case BANNER_INVALID:
```

considering it's an enum type that's being switched on.